glossary.md

```diff
@@ -50,6 +50,7 @@ Nếu bạn cho rằng một từ không nên dịch ra tiếng Việt, bạn c
 | Blackbox dev set        | tập phát triển Blackbox        | [https://git.io/JvQx3](https://git.io/JvQx3) |
 | bounding box            | khung chứa                     | [https://git.io/JvQxs](https://git.io/JvQxs) |
 | broadcast               | lan truyền                     | [https://git.io/Jvoj3](https://git.io/Jvoj3) |
+| beam search             | tìm kiếm beam                  |                                              |
```

Comment:
mình vừa thấy ở sách [MLY](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/blob/master/chapters/book_vn.md#44-b%C3%A0i-ki%E1%BB%83m-tra-x%C3%A1c-minh-t%E1%BB%91i-%C6%B0u) cũng dịch là `tìm kiếm chùm tia` nè
![image](https://user-images.githubusercontent.com/34294427/84393992-88ad1280-ac26-11ea-86ce-ac72adee009b.png)

Reply:
Vậy hả, sách đấy cũng AIVN mình dịch mà :)) Nhưng mình thấy giáo trình học thuật họ đã dịch sẵn là `tìm kiếm beam` thì nên giữ nguyên. Anw, đây là cơ hội để thảo luận thêm về từ này (có thể lần trc chưa thảo luận kĩ). 
PS: Chùm tia nghe vật lý quá mà không thể hiện rõ bản chất lắm.

Reply:
ừm mình vừa search thì cũng chỉ có 1 2 chỗ dịch là `chùm`, còn hầu hết giáo trình dịch là `tìm kiếm beam`

Reply:
Yeah, Phần @nguyenduydu dịch thì sử dụng từ này. Đợi xem ý kiến mn thế nào rồi chúng ta thống nhất từ này cho nhất quán ;)

Reply:
@davidnvq anh vui lòng kèm thêm reference được không ạ?

Reply:
Em GG r thêm vào hộ anh nhé, a search ra 1 đống giáo trình từ các trường đh. A ngủ trước đây, mai có pv :'(

Reply:
Mình vote là `chùm tia` do từ beam ở đây được xài với nghĩa thông dụng của nó, cũng không có gì sợ nhầm lẫn.

Reply:
Từ này khá thú vị nhỉ, nhận được đóng góp ý kiến của mn rất hay :3

Mình hoàn toàn nhất trí với @lkhphuc rằng từ "tìm kiếm chùm tia" thì sẽ k lo nhầm lẫn vì không còn ngữ cảnh nào khác của từ này trong IT. Tuy vậy mình lại k đồng tình với việc dịch như thế, để mình đưa ra 1 vài điểm ủng hộ việc giữ nguyên cách dịch như trước đây là "tìm kiếm beam":

Ưu:
1. Đã có rất nhiều tài liệu, sách giáo trình, bài giảng, v.v., (ví dụ: sách trí tuệ nhân tạo của NXB ĐHSPHN, sách trí tuệ nhân tạo của GS Nguyễn Thanh Thuỷ ĐHBKHN năm 2007 gì đó, và các sách về giải thuật nói chung) đã dịch "tìm kiếm beam". Nếu người đọc của D2L muốn tìm hiểu thêm, rất đơn giản là copy paste cụm từ này vào GG thì ra vô vàn kết quả. Ngược lại, không có ví dụ (at least GG search) support cho "tìm kiếm chùm tia"  ngoại trừ 1 em SV trường mình viết báo cáo dịch là như vậy.
2. Từ "tìm kiếm beam" ngắn gọn và cũng k nhầm lẫn với bất cứ thuật ngữ nào nữa (so far). 
3. Không chỉ những ví dụ tài liệu nêu trên mà có rất nhiều bài giảng, bài viết của giảng viên, ng làm nghiên cứu họ kế thừa thuật ngữ này. Chứng tỏ nó đã được qua được bài "test of time". Hẳn cũng có lý do.
Nhược:
1. Không dịch hẳn ra tiếng Việt dù từ beam có trong từ điển kỹ thuật và ở đây cũng hiểu na ná là "chùm" k "tia" tốt nhất?

Vì họ cũng k giải thích tại sao lại giữ cách dịch như vậy nên mình trình bày cách hiểu chủ quan một chút. 
1. Hiển nhiên từ `beam` hoàn toàn có nghĩa trong Vật lý, Kỹ thuật xây dựng, toán nên mình không nghĩ là những người sau họ "lười" suy nghĩ mà sao y cách cũ.  Tuy nhiên, không có cách dịch nào cho thuật ngữ `beam` trong Tin học. Thật sự là các thầy cô và học giả cũng đã rất sáng tạo khi dịch được những thuật ngữ như "tìm kiếm leo đồi", "tìm kiếm mô phỏng luyện kim", v.v.

2. Các thuật toán tìm kiếm (tìm kiếm đường đi, tìm kiếm câu tốt nhất, v.v.) đều được xây dựng dựa trên cây tìm kiếm. Trong các thuật ngữ liên quan tới cây, "nhánh, nút, lá, gốc, v.v." lại không có thuật ngữ nào dịch là "chùm" và "tia". Nếu có thì nghe cũng không tương quan lắm. Quay lại "beam search", ở mỗi timestep, mình sample và store k-outputs tốt nhất (output ở đây tương đương với nút trong cây, và ở hiện tại thì nó có thể coi là `lá` của cây được sinh (so far)). "Tia" ở đây mượn từ bên vật lý nhưng lại k khớp vào "cây". "Chùm" cũng make sense đôi chút, phải chăng nên dịch là "chùm nút hay chính xác hơn là "chùm lá"?  Bản thân từ "beam" có thể dịch là "chùm"/"tia"/"dầm" chứ dịch là "chùm tia" thì lại càng ít phổ biến.

Mình ủng hộ giữ nguyên, một phần vì người đọc, một phần tôn trọng cách dịch của học giả trước, và một phần cũng thấy nó bất hợp lý khi dịch hết cả từ :3
Anw, mn cứ thảo luận thoải mái nhé, chúng ta hoàn toàn có quyền phản biện lại cách dịch của ng đi trước mặc cho học vị họ thế nào :)))

Reply:
@rootonchair: Sorry e, tối qua muộn quá phải ngủ sớm vì sáng có chút việc nên hơi đứt quãng. Ví dụ này khá hiển nhiên nên a đưa link đại diện em tham khảo nha ;)) https://bit.ly/2UAOe22

Reply:
Thank @davidnvq , học thêm được nhiều từ khá thú vị (mô phỏng luyện kim :)).

>     1. Đã có rất nhiều tài liệu, sách giáo trình, bài giảng, v.v., (ví dụ: sách trí tuệ nhân tạo của NXB ĐHSPHN, sách trí tuệ nhân tạo của GS Nguyễn Thanh Thuỷ ĐHBKHN năm 2007 gì đó, và các sách về giải thuật nói chung) đã dịch "tìm kiếm beam". Nếu người đọc của D2L muốn tìm hiểu thêm, rất đơn giản là copy paste cụm từ này vào GG thì ra vô vàn kết quả. Ngược lại, không có ví dụ (at least GG search) support cho "tìm kiếm chùm tia"

Không phải bắt bẻ gì nhưng mình search trên google.com.vn `tìm kiếm beam` thì trên trang nhất chỉ có duy nhất một kết quả tiếng việt là trang này (https://voer.edu.vn/m/cac-chien-luoc-tim-kiem-kinh-nghiem/d478b017).

Như trên trang đó có ghi, hầu hết các tên trong `Giải thuật và lập trình` đều được dịch đầy đủ ra tiếng việt, nên mình rất không hiểu lý do gì từ `beam search` trước đây không được dịch. Có thể là một người xài đầu tiên không dịch và thành ra mọi người theo sau đều không dịch? Hoặc từ `beam` có một nghĩa đặc biệt nào trong trường hợp này mình chưa thấy?


>     2\. Quay lại "beam search", ở mỗi timestep, mình sample và store k-outputs tốt nhất (output ở đây tương đương với nút trong cây, và ở hiện tại thì nó có thể coi là `lá` của cây được sinh (so far)). "Tia" ở đây mượn từ bên vật lý nhưng lại k khớp vào "cây". "Chùm" cũng make sense đôi chút, phải chăng nên dịch là "chùm nút hay chính xác hơn là "chùm lá"?  Bản thân từ "beam" có thể dịch là "chùm"/"tia"/"dầm" chứ dịch là "chùm tia" thì lại càng ít phổ biến.

Chỗ này mình cũng hơi lấn cấn trước đây. Giờ đọc thấy ý kiến này thì đúng là `chùm tia` chưa phù hợp lắm. 

>  Tuy nhiên, không có cách dịch nào cho thuật ngữ beam trong Tin học. 

Mình nghĩ không có cách dịch trong tin học vì nó không mang ý nghĩa đặc biệt gì khác, từ `beam` vẫn mang ý nghĩa `chùm` thông thường => Mình thấy `tìm kiếm chùm` vừa tổng quát lại chính xác.

Reply:
Anyway ace vote có thích dịch hay không:

* 👍 `tìm kiếm beam`
* ❤️ `tìm kiếm chùm`

Reply:
Khi dịch phần này mình cũng đã tìm xem có tài liệu nào dịch chưa. Ngoài tài liệu @lkhphuc chỉ ra bên trên, mình thấy giáo trình Trí Tuệ Nhân Tạo của Học Viện Kỹ Thuật Quân Sự cũng dịch là `tìm kiếm chùm`. 

https://fit.lqdtu.edu.vn/files/FileMonHoc/1.%20TTNT.pdf

Reply:
@davidnvq nếu ông giáo không còn phản biện thì mình chốt từ này. Chờ ông giáo thêm 1 hôm nữa nhé.

Reply:
Thanks cả nhà đã discuss sôi nổi :3 bây giờ mới onl được. Mình vẫn ủng hộ `tìm kiếm beam` 

> Không phải bắt bẻ gì nhưng mình search trên google.com.vn `tìm kiếm beam` thì trên trang nhất chỉ có duy nhất một kết quả tiếng việt là trang này (https://voer.edu.vn/m/cac-chien-luoc-tim-kiem-kinh-nghiem/d478b017).

Sorry lỗi này là ở mình dù có đưa nguồn sách kia nhưng chắc mọi người search k ra do thiếu từ khoá "tìm kiếm beam trí tuệ nhân tạo" trong link trl em Vinh.
Mình vote từ "tìm kiếm beam" bởi vì mình nhớ đã gặp nó trong giáo trình TTNT (Nguyễn Thanh Thuỷ) mà mình được học, khá tiếc là k tìm thấy bản mềm trên mạng của cuốn sách đó.

Mình xin phép trích dẫn lại từng giáo trình hoặc tài liệu khác mà available sẵn từ GG:

http://ccs1.hnue.edu.vn/hoanpt/AI/Tailieu-TriTueNhanTao_new1.pdf
https://kenhdaihoc.net/threads/giao-trinh-tri-tue-nhan-tao-dhqg-ha-noi.19164/

https://fit.lqdtu.edu.vn/files/FileMonHoc/1.%20TTNT.pdf
https://users.soict.hust.edu.vn/quangnn/ai/slides/L4-Tim_kiem_heuristic.pdf (giữ nguyên cả cụm)
https://fit.mta.edu.vn/files/FileMonHoc/1.%20DC%20CT%20Trf%20tu_%20nhGn%20t_o.pdf


Slide này cũng khá mang tính cá nhân của người dạy, tuy nhiên chỉ có 1 từ "tìm kiếm chùm" trong slide, còn thầy này sử dụng 3 từ tìm kiếm beam trong slide :)) ???
> Mình thấy giáo trình Trí Tuệ Nhân Tạo của Học Viện Kỹ Thuật Quân Sự cũng dịch là `tìm kiếm chùm`.
> https://fit.lqdtu.edu.vn/files/FileMonHoc/1.%20TTNT.pdf


Mình thắc mắc lại câu trả lời của Phúc:

> Mình vote là `chùm tia` do từ beam ở đây được xài với nghĩa thông dụng của nó, cũng không có gì sợ nhầm lẫn.
"Thông dụng" ở đây là thông dụng trong lĩnh vực nào, rõ ràng chưa có tài liệu chính thống nào mình search được trên GG sử dụng "tìm kiếm chùm (tia)" ngược lại thì có rất nhiều sách giáo trình đã sử dụng từ này. 

Dù số đông vote "thích" ở đây nhưng mình nghĩ final thought cũng nên count cả những học giả trước nữa :)))

Reply:
> từ beam vẫn mang ý nghĩa chùm thông thường => Mình thấy tìm kiếm chùm vừa tổng quát lại chính xác.

Thêm một cmt nữa nếu Phúc có thể làm rõ hơn từ "tổng quát" là trong TH nào và chính xác là trong trường hợp nào? Mình thì thấy hơi cấn khi ghép từ "chùm" vào 1 cây. Có chùm nho, chùm quả, v.v. chứ rất ít khi nghe chùm cành, chùm lá, chùm nút, v.v. Và cũng từ cách hiểu từ chùm như vậy nên ta nghĩ rằng phần tử thuộc chùm sẽ có cùng cha (ví dụ chùm nho thì sẽ có cùng cuống). Trong search beam, thực tế là không hẳn 100% top-K các node (nút hoặc lá với cây được sinh so far) không luôn luôn có cùng nút cha. Nên "chính xác" thì mình thấy chưa đúng lắm với "search beam" khi dịch ra tiếng Việt.

Reply:
> chứ rất ít khi nghe chùm cành, chùm lá, chùm nút, v.v. 

Mình lại thấy các từ này hoàn toàn bình thường. Một điều mình nhận thấy qua nhiều thảo luận nhiều thuật ngữ khác nhau là đa phần lý do mọi người bàn cãi nhiều là vì khác biệt hiểu biết trong tiếng việt chứ không ở bản thân thuật ngữ tiếng anh. Có lẽ ta đã tới mức này với từ beam :).

> Trong search beam, thực tế là không hẳn 100% top-K các node (nút hoặc lá với cây được sinh so far) không luôn luôn có cùng nút cha. Nên "chính xác" thì mình thấy chưa đúng lắm với "search beam" khi dịch ra tiếng Việt.

 Như trong trường hợp này, nếu mình đi đủ xa thì tất cả các nhánh đều cũng sẽ có chung một gốc thôi đúng không, tất cả các từ được sinh ra đều có chung gốc token `bos - beginning of sequence`. Mình thấy không nên đi quá sâu vào hình ảnh liên tưởng của một thuật ngữ vì như trên, cùng một từ nhưng hình ảnh liên tưởng chỉ tiết của mỗi người một khác. Mình chỉ cần nắm bắt được hình ảnh chung và tổng quát là được. 

> Dù số đông vote "thích" ở đây nhưng mình nghĩ final thought cũng nên count cả những học giả trước nữa :))) 
Thêm một cmt nữa nếu Phúc có thể làm rõ hơn từ "tổng quát" là trong TH nào và chính xác là trong trường hợp nào?

"Thông dụng và tổng quát" ý mình chính xác là nó không có mang một ý nghĩa đặc biệt nào trong ngữ cảnh này cả. 
Giông như `hill climbing` được dịch là `thuật toán leo đồi`, `hill` ở đây mang nghĩa thông dụng là `đồi núi`, không có ý nghĩa gì đặc biệt khác trong. Tương tự `beam` ở đây chỉ một `chùm`, tức một nhóm nhỏ, ko có ý nghĩa gì đặc biệt trong trường hợp này.

Mình hiểu là nếu đã được học qua với từ `tìm kiếm beam` thì sẽ thấy rất kì khi nó được dịch qua tiếng việt, nhưng tiêu chí của dự án mình là luôn cố dịch nhiều nhất có thể, chỉ vay mượn khi không có từ nào có thể thay thế.

Thank @davidnvq đã có nhưng chia sẻ rất thú vị. Tới đây mình nghĩ là ta đã hiểu ý nhau hết rồi, và cũng tin rằng không ai có thể thuyết phục ai được nữa nên xin gác lại tại đây và chờ @duythanhvn cho phán xét cuối cùng.

Reply:
Mình xin tổng kết từ này là **tìm kiếm chùm** anh em nhé. 

Trọng điểm của dự án khi thảo luận những thuật ngữ khi không có từ thay thế mới không dịch từ này sang tiếng Việt. Mình cũng thông tin với anh em là ở đây, chúng ta giới hạn từ này (và những thuật ngữ được thảo luận trước đó cùng những thuật ngữ sẽ được thảo luận trong tương lai) được dịch để sử dụng riêng cho tài liệu này, cuốn sách này; vì vậy mà nếu nó phục vụ được cho mục tiêu đó mà không có bất kỳ vấn đề gì khác thì mình không có lý do để mà không dịch.

Nếu ở những tài liệu được dẫn nguồn trên giữ nguyên mà ta không chỉ ra được bất kỳ lý do vì sao họ lại giữ nguyên (ngoài lý do cố hữu là vì tôi không tìm được từ phù hợp thì tôi không dịch ra), thì ở bản thân mình cũng không có lý do gì để không đề xuất cách dịch này nhằm có sự thông dụng mới trong việc sử dụng thuật ngữ này trong hiện tại.

Hiện tại mình tạm chốt thuật ngữ này như thế này đã nhé. Ta có thể thảo luận thêm khi ta có thêm nguồn lực, thêm hiểu biết, thêm góc nhìn anh em nha.